SimplifiedTests/NYPLAxisLicenseServiceTests.swift

```diff
@@ -0,0 +1,204 @@
+      description: "Invalid license should not be validated")
+    failedValidationExpectation.expectedFulfillmentCount = 2
+    
+    let license = createLicenseService(bookVaultId: TestBook.book1.bookVaultId,
+                                       cypher: cypherReturningBook2VaultId,
+                                       isbn: TestBook.book1.isbn)
+    
+    let licenseURL = Bundle(for: NYPLAxisLicenseServiceTests.self)
+      .url(forResource: "license-\(license.isbn)", withExtension: "json")!
+    let savedLicenseURL = licenseHomeDirectory
+      .appendingPathComponent(axisKeysProvider.desiredNameForLicenseFile)
+    if FileManager.default.fileExists(atPath: savedLicenseURL.path) {
+      try! FileManager.default.removeItem(at: savedLicenseURL)
+    }
+    
+    let licenseData = try! Data(contentsOf: licenseURL)
+    try! NYPLAssetWriter().writeAsset(licenseData, atURL: savedLicenseURL)
+    XCTAssertTrue(FileManager.default.fileExists(atPath: savedLicenseURL.path))
+    
+    itemDownloader.downloadsTerminated = {
+      failedValidationExpectation.fulfill()
+    }
+    
+    license.didLogLicenseError = {
+      if $0 == "AxisLicenseService validation failed due to invalid vault ID in license" {
+        failedValidationExpectation.fulfill()
+      }
+    }
+    license.validateLicense()
+    
+    wait(for: [failedValidationExpectation], timeout: 10.0)
+  }
+  
+  func testLicenseWithValidKeycheckShouldBeValidated() {
+    let validationExpectation = XCTestExpectation(
+      description: "Valid license should be validated")
+    
+    let license = createLicenseService(bookVaultId: TestBook.book2.bookVaultId,
+                                       cypher: cypherReturningBook2VaultId,
+                                       isbn: TestBook.book2.isbn)
+    
+    let licenseURL = Bundle(for: NYPLAxisLicenseServiceTests.self)
+      .url(forResource: "license-\(license.isbn)", withExtension: "json")!
+    let savedLicenseURL = licenseHomeDirectory
+      .appendingPathComponent(axisKeysProvider.desiredNameForLicenseFile)
+    if FileManager.default.fileExists(atPath: savedLicenseURL.path) {
+      try! FileManager.default.removeItem(at: savedLicenseURL)
+    }
+    
+    let licenseData = try! Data(contentsOf: licenseURL)
+    try! NYPLAssetWriter().writeAsset(licenseData, atURL: savedLicenseURL)
+    XCTAssertTrue(FileManager.default.fileExists(atPath: savedLicenseURL.path))
+    
+    license.validateLicense()
+    itemDownloader.dispatchGroup.notify(queue: .global()) {
+      if self.itemDownloader.shouldContinue {
+        validationExpectation.fulfill()
+      }
+    }
+    wait(for: [validationExpectation], timeout: 10.0)
+  }
+  
+  func testLicenseShouldSaveBookInfoForLaterUse() {
+    let savedInfoExpectation = XCTestExpectation(
+      description: "Book info should be saved for later use")
+    
+    let license = createLicenseService(bookVaultId: TestBook.book1.bookVaultId,
```

Comment:
can you rename `license` to `service`? if I understand correctly that's what we are creating here...

Reply:
That makes more sense. Done.